### PR TITLE
chore: pass playwright.env config to all node calls

### PR DESCRIPTION
**Pull request by chasestarr**
This will support pnp package managers (yarn 2, pnpm) by allowing `NODE_OPTIONS` to be configured within `playwright.env` and passed to node calls. For example, when using yarn 2 I can edit my `.vscode/settings.json` file to below and register the pnp.cjs file.

```json
{
  "playwright": {
    "env": {
      "NODE_OPTIONS": "-r ./.pnp.cjs"
    }
  }
}
```

### CODE REVIEW

**Comment by chasestarr** [comment]
@microsoft-github-policy-service agree company="Uber"

**Comment by chasestarr** [comment]
Hi, checking in again. Is this something the team is interested in reviewing?

File: src/playwrightTest.ts [hunk 1]
```diff
@@ -71,12 +71,11 @@ export class PlaywrightTest {
...
-        'try { const coreIndex = require.resolve("playwright-core"); console.log(JSON.stringify({ coreIndex })); } catch { console.log("undefined"); }',
-      ], path.dirname(pwtIndex));
-      const { coreIndex } = JSON.parse(coreInfo);
-      let cli = path.resolve(coreIndex, '..', 'lib', 'cli', 'cli');
+        'try { const cli = require.resolve("@playwright/test/cli"); console.log(JSON.stringify({ cli })); } catch { console.log("undefined"); }',
+      ], path.dirname(configFilePath), settingsEnv);
+      let { cli } = JSON.parse(cliInfo);
 
       // Dogfood for 'ttest'
       if (cli.includes('packages/playwright-core') && configFilePath.includes('playwright-test'))
```
Comment (mxschmitt) [inline src/playwrightTest.ts:81]: this dogfood seems broken after this change. We used it when we are inside our repository (microsoft/playwright). 

```suggestion
      if (cli.includes('packages/playwright-test') && configFilePath.includes('playwright-test'))
```
it should work after that.

**Comment by mxschmitt** [comment]
Thanks for your PR, after the requested changes we're happy to add it.

**Comment by chasestarr** [comment]
Resolved the change requests. Thanks!

**Review by pavelfeldman — APPROVED** [review]

**Pull request merged**